src/BinSkim.Driver/AnalyzeCommand.cs

```diff
@@ -261,58 +272,122 @@ private HashSet<IBinarySkimmer> CreateSkimmers(IMessageLogger<BinaryAnalyzerCont
+                            disabledSkimmers.Add(skimmer.Id);
+                            break;
+                        }
 
                         case AnalysisApplicability.ApplicableToSpecifiedTarget:
+                        {
+                            try
                             {
-                                try
-                                {
-                                    skimmer.Analyze(context);
-                                }
-                                catch (Exception ex)
-                                {
-                                    LogUnhandledRuleExceptionAnalyzingTarget(disabledSkimmers, context, skimmer, ex);
-                                }
-                                break;
+                                skimmer.Analyze(context);
                             }
+                            catch (Exception ex)
+                            {
+                                LogUnhandledRuleExceptionAnalyzingTarget(disabledSkimmers, context, skimmer, ex);
+                            }
+                            break;
+                        }
                     }
                 }
-                DisposeContext(context);
+
+                // Once we've processed all portable executable skimmers for a specific
+                // target, we'll proactively let go of the data associated with this 
+                // analysis phase. Follow-on analyses (such as the Roslyn integration)
+                // shouldn't attempt to rehydrate this data. The context implementation
+                // currently raises an exception if there is an attempt to rehydrate a
+                // previously nulled PE instance.
+                DisposePortableExecutableContextData(context);
+
+                // IsManagedAssembly is computed on intitializing the binary context
+                // object and is still valid after disposing the PE data. The Roslyn
+                // analysis is driven solely off the binary file path in the context.
+                if (context.IsManagedAssembly && roslynAnalyzerFilePaths?.Count > 0)
+                {
+                    AnalyzeManagedAssembly(context.Uri.LocalPath, roslynAnalyzerFilePaths, context);
+                }
             }
         }
 
-        private static void DisposeContext(BinaryAnalyzerContext context)
+        private static void DisposePortableExecutableContextData(BinaryAnalyzerContext context)
         {
-            if (context.PE != null) { context.PE.Dispose(); }
-            // TODO: handle pdb but we don't want to fault the PDB in a 2nd time           
+            context.DisposePortableExecutableData();
+        }
+
+        private void AnalyzeManagedAssembly(string assemblyFilePath, IEnumerable<string> roslynAnalyzerFilePaths, BinaryAnalyzerContext context)
+        {
+            ILDiagnosticsAnalyzer roslynAnalyzer = new ILDiagnosticsAnalyzer();
+
+            if (_globalRoslynAnalysisContext == null)
```

Comment:
ok, now we retain a global analysis context object that is initialized once only. we push this onto new ILDiagnosticsAnalyzer instances rather than reloading from analyzer paths. Did I get all this right? I will defer the end compilation action to a future change: because this callback can report diagnostics, I will need to plumb through a mechanism to log through typical channels.